module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,656 @@
+
+                    std::vector<char> data =
+                        NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(to_send_now);
+                    // Size of the message, in network endian
+                    uint32_t Nn = htonl(data.size());
+                    // Send the message size first
+                    if (send(fd, &Nn, sizeof(Nn), 0) != sizeof(Nn)) {
+                        log<NUClear::ERROR>(
+                            fmt::format("Error in sending ActuatorRequests' message size,  {}", strerror(errno)));
+                    }
+                    // then send the data
+                    if (send(fd, data.data(), data.size(), 0) != (signed) data.size()) {
+                        log<NUClear::ERROR>(fmt::format("Error sending ActuatorRequests message, {}", strerror(errno)));
+                    }
+
+                    log<NUClear::TRACE>("Sending actuator request.");
+                });
+
+        error_io = on<IO>(fd, IO::CLOSE | IO::ERROR).then([this, server_address, port](const IO::Event& /*event*/) {
+            // Something went wrong, reopen the connection
+            setup_connection(server_address, port);
+        });
+
+        shutdown_handle = on<Shutdown>().then([this] {
+            // Disconnect the fd gracefully
+            if (fd != -1) {
+                shutdown(fd, SHUT_RDWR);
+                close(fd);
+                fd = -1;
+            }
+        });
+
+        // Send initial message to activate the servos
+        std::vector<char> data =
+            NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(make_inital_acutator_request(time_step));
+
+        uint32_t Nn = htonl(data.size());
+
+        // Send the message size first
+        if (send(fd, &Nn, sizeof(Nn), 0) != sizeof(Nn)) {
+            log<NUClear::ERROR>(fmt::format("Error in sending ActuatorRequests' message size,  {}", strerror(errno)));
+        }
+        // then send the data
+        if (send(fd, data.data(), data.size(), 0) != (signed) data.size()) {
+            log<NUClear::ERROR>(fmt::format("Error sending ActuatorRequests message, {}", strerror(errno)));
+        }
+
+        // Temporary measure to show that we can pass servo targets
+        // log("sending hardcoded servotargets to robot to test");
+        // auto target     = message::motion::ServoTarget();
+        // target.time     = NUClear::clock::now();
+        // target.id       = 2;
+        // target.position = 2;
+        // target.gain     = 10;
+        // target.torque   = 10;
+        // auto targets    = std::make_unique<ServoTargets>();
+        // targets->targets.push_back(target);
+        // emit(targets);
+    }
+
+    void Webots::translate_and_emit_sensor(const SensorMeasurements& sensor_measurements) {
+        if (false) {  // Change to true to print the received sensor_measurements, will remove after this is done
```

Comment:
Maybe just change all of these to `log<NUClear::TRACE>()` statements? Optionally, also wrap them in a `#ifndef NDEBUG ... #endif` block?

Reply:
the ifdef is handles by the if(false) statement.. and i dont care for changing it :P